# Binomial

In [ ]:
import scipy.stats as stats
import numpy as np

A fair coin is flipped 10 times. What is the probability of getting exactly 6 heads?

In [7]:
stats.binom.pmf(6, 10, 0.5)

np.float64(0.2050781249999999)

---
A basketball player makes free throws with probability 𝑝 = 0.75. She takes 8 shots. What is the probability she makes at least 7 shots?

In [10]:
1 - stats.binom.cdf(6, 8, 0.75)

np.float64(0.3670806884765625)

---
A machine produces items that are defective with probability 0.02. A batch contains 200 items.
- What is the expected number of defective items?
- What is the variance?

In [11]:
expected = 0.02 * 200
variance = 200 * 0.02 * 0.98
print(f"expected = {expected}")
print(f"variance = {variance}")

expected = 4.0
variance = 3.92


---
A drug works for 90% of patients. In a group of 12 patients, what is the probability the drug works for at most 10 of them?

In [12]:
stats.binom.cdf(10, 12, .9)

np.float64(0.34099774821099965)

# Poisson
A call center receives an average of 𝜆=5 calls per hour. What is the probability they receive exactly 3 calls in the next hour?

In [13]:
stats.poisson.pmf(3, 5)

np.float64(0.1403738958142805)

---
Traffic accidents occur at a rate of 2 per day. What is the probability that no accidents occur in the next 12 hours?

In [14]:
lam = 2 * 0.5
stats.poisson.pmf(0, lam)

np.float64(0.36787944117144233)

---
A website gets an average of 0.4 server errors per hour. What is the probability that at least one error occurs in the next hour?

In [15]:
stats.poisson.sf(0, 0.4)

np.float64(0.3296799539643607)

---
Which of the following is appropriate for a Poisson model?
- a) Number of raindrops hitting a window per minute
- b) Number of correct answers on a multiple‑choice test
- c) Number of customers arriving at a store per hour
- d) Number of defective items in a fixed batch of 100

Ans: a, c

---
A bookstore sees an average of 18 customers per hour. What is the probability that 40 or more customers arrive in a 2‑hour period?

In [16]:
lam = 18 * 2
stats.poisson.sf(39, lam)

np.float64(0.2736963932046033)

# Normal Distribution
Heights of adult men are normally distributed with mean 𝜇=178 cm and standard deviation 𝜎=7 cm. What is the z‑score of a man who is 190 cm tall?

In [20]:
z = (190 - 178) / 7
z

1.7142857142857142

#### Answer
$ z = \frac{x - \mu}{\sigma} $

$ z = \frac{190 - 178}{7} $

$ z = 1.7142857142857142 $

A man who is 190cm tall has a z-score of about 1.71, meaning he is 1.71 standard deviations above the mean.

---

Using the same distribution as above:

What proportion of men are between 170 cm and 185 cm?

In [21]:
stats.norm.cdf(185, loc=178, scale=7) - stats.norm.cdf(170, loc=178, scale=7)

np.float64(0.7147957915949852)

---
IQ scores are normally distributed with mean 100 and standard deviation 15. What percentage of people have an IQ above 130?

In [22]:
stats.norm.sf(130, loc=100, scale=15)

np.float64(0.0227501319481792)

SAT math scores are normally distributed with mean 530 and standard deviation 100. What score corresponds to the 90th percentile?

In [25]:
ans_z_table = (1.28 * 100) + 530

mu = 530
std = 100
p = 0.9
ans_ppf = stats.norm.ppf(p, loc=mu, scale=std)
print(f"Answer using z-table: {ans_z_table}")
print(f"Answer using ppf: {ans_ppf}")

Answer using z-table: 658.0
Answer using ppf: 658.1551565544601


---
A factory produces lightbulbs with a 3% defect rate. A shipment contains 800 bulbs.  Use a normal approximation to estimate the probability that fewer than 20 bulbs are defective.

---
# Geometric Distribution

Suppose 𝑋∼Geom(𝑝). What is the memoryless property, and how does it apply to 𝑃(𝑋>10∣𝑋>5)?

### Answer
The memorlyess property is formally written as:
$$
P(X > m+n | X > m) = P(X > n)
$$

Thus $ m=5 $ and $ n=5 $ so $ P(X > 10 | X > 5) = P(X < 5) = (1-p)^5
Thus, once 5 failures have already occurred, the probability of needing more than 10 trials is the same as the probability of needing more than 5 additional trials — as if the first 5 trials never happened.r.

---
A basketball player makes each free throw withprobability 𝑝=0.7.Let 𝑋 be the number of shots until she makes her firstbasket. Compute:
- (a) 𝑃(𝑋=1)
- (b) 𝑃(𝑋=3)
- (c) 𝑃(𝑋>3)

In [1]:
import scipy.stats as stats

print(f"P(X=1) {stats.geom.pmf(1, 0.7)}")
print(f"P(X=3) {stats.geom.pmf(3, 0.7)}")
print(f"P(X>3)) {stats.geom.sf(3, 0.7)}")

P(X=1) 0.7
P(X=3) 0.06300000000000001
P(X>3)) 0.027000000000000007


---
A machine produces a defective part with probability 𝑝=0.02. Let 𝑋 be the number of parts inspected until the first defective one appears. Compute:
- (a) 𝑃(𝑋=10)
- (b) 𝑃(𝑋>50)
- (c) 𝐸[𝑋]

In [2]:
import scipy.stats as stats

p=0.02
print(f"P(X=10) {stats.geom.pmf(10, p)}")
print(f"P(X>50) manual: {(1-p)**50}")
print(f"P(X>50) {stats.geom.sf(50, p)}")

P(X=10) 0.016674955242602995
P(X>50) manual: 0.36416968008711675
P(X>50) 0.3641696800871171


The $E[X]$ depends on the convention used to derive the geometric distribution.  There are 2 definitions that describe the same process.
- The number of trials until the first success: $ E[X] = \frac{1}{p} $
- The number of failures until the first success: $ E[X] = \frac{1-p}{p} $

---
Let 𝑋∼Geom(0.4). Compute:
- (a) 𝑃(𝑋≤3)
- (b) 𝑃(𝑋>5)
- (c) Var(𝑋)

In [6]:
import scipy.stats as stats
p=0.4
print(f"P(X<=3) {stats.geom.cdf(3, p)}")
print(f"P(X>4) {stats.geom.sf(5, p)}")
print(f"Var(X) = {(1-p)/p**2}")

P(X<=3) 0.784
P(X>4) 0.07776
Var(X) = 3.749999999999999


# Exponential Distribution
